components/esp_wifi_remote/Kconfig.soc_wifi_caps.in

```diff
@@ -224,14 +224,46 @@ if SLAVE_IDF_TARGET_ESP32C6
 
 endif # ESP32C6
 
-if SLAVE_IDF_TARGET_ESP32H2
+if SLAVE_IDF_TARGET_ESP32C5
 
-endif # ESP32H2
-
-if SLAVE_IDF_TARGET_ESP32P4
+    config SLAVE_SOC_WIFI_SUPPORTED
+        bool
+        default y
 
     config SLAVE_SOC_WIFI_LIGHT_SLEEP_CLK_WIDTH
         int
         default 12
 
-endif # ESP32P4
+    config SLAVE_SOC_WIFI_HW_TSF
+        bool
+        default y
+
+    config SLAVE_SOC_WIFI_FTM_SUPPORT
+        bool
+        default n
+
+    config SLAVE_SOC_WIFI_GCMP_SUPPORT
+        bool
+        default y
+
+    config SLAVE_SOC_WIFI_WAPI_SUPPORT
+        bool
+        default y
+
+    config SLAVE_SOC_WIFI_CSI_SUPPORT
+        bool
+        default y
+
+    config SLAVE_SOC_WIFI_MESH_SUPPORT
+        bool
+        default y
+
+    config SLAVE_SOC_WIFI_HE_SUPPORT
+        bool
+        default y
+
+    config SLAVE_SOC_WIFI_HE_SUPPORT_5G
```

Comment:
There will still be very much confusion despite of this flag. The iperf examples use non slave flags.

For example,
![Screenshot 2024-09-13 at 4 42 12 PM](https://github.com/user-attachments/assets/703cba4a-1934-4d99-86f4-8f604d9a7a91)


Hosted would be aware, but i think we need to translate slave flags to normal ones for applications?

Reply:
This was probably the most important design decision that we made.

> There will still be very much confusion despite of this flag.

which is I think expected, architecturally, as wifi-remote provides `SLAVE_SOC...` configs based on the **remote** targets. If we think of a "board" with both host-slave targets (super-target) we could do that, maybe, but not on wifi-remote layer.

>  The iperf examples use non slave flags.

Yes, that's an example/project, that need to use host side (or maybe the "super-target" side?)

---

One idea of a structurally correct solution would be to come up with `BSP_WIFI_..._SUPPORT` macros, which would combine host capabilities with slave capabilities.

Reply:
1. I am not against the design decision. In fact it is correct to be done that way.

I was actually pointing to practical problem, that, the examples are written without consideration of these flags aware. Problem with that is the example code is going to simply discard the SOC feature to be not supported.

For example, in prior comment, ideally example should have added OR with 
SLAVE_SOC_WIFI_HE_SUPPORT_5G.

But examples simply assume the target (for example P4) is checked for itself. ideally P4 should inherit the SOC features then it would be very transparent . What would you suggest?

Options would be :
- 1. examples to change
- 2. expose same SOC feature flag
- 3. something else?

2. Also CONFIG_WIFI_CMD_ENABLE_BAND is also inherited in some way for C5 as slave?

Reply:
1) I've vote for updating the example -- the wifi-cmd component in this case. This would also be in line with what we discussed with jack when splitting wifi functionality into native/local. All API would be available, some would just not be supported in some cases.
From the example/project perspective it should be transparent if we call `set_wifi_bands()` on C5 or P4.

2) Yes, they're populated from wifi-remote Kconfig based on `SLAVE_SOC_WIFI_...` capabilities

Reply:
Sure, then, some new MR might be required to add such slave aware changes for examples throughout.
I will rest the case for now.